src/assets.zig

```diff
@@ -42,76 +85,116 @@ fn readDefaultFile(allocator: NeverFailingAllocator, dir: std.fs.Dir) !ZonElemen
 	return .null;
 }
 
-/// Reads .zig.zon files recursively from all subfolders.
-pub fn readAllZonFilesInAddons(externalAllocator: NeverFailingAllocator, addons: main.List(std.fs.Dir), addonNames: main.List([]const u8), subPath: []const u8, defaults: bool, output: *std.StringHashMap(ZonElement)) void {
+/// Reads all asset `.zig.zon` files recursively from all sub folders.
+///
+/// Files red are stored in output hashmap with asset ID as key.
+/// Asset ID are constructed as `{addonName}:{relativePathNoSuffix}`.
+/// relativePathNoSuffix is always unix style path with all extensions removed.
+pub fn readAllZonFilesInAddons(
+	externalAllocator: NeverFailingAllocator,
+	addons: main.List(std.fs.Dir),
+	addonNames: main.List([]const u8),
+	assetSubdirectory: []const u8,
+	defaults: bool,
+	output: *std.StringHashMap(ZonElement),
+	migrations: ?*std.StringHashMap(ZonElement),
+) void {
 	for(addons.items, addonNames.items) |addon, addonName| {
-		var dir = addon.openDir(subPath, .{.iterate = true}) catch |err| {
-			if(err != error.FileNotFound) {
-				std.log.err("Could not open addon directory {s}: {s}", .{subPath, @errorName(err)});
+		var addonAssetsSubdirectory = addon.openDir(assetSubdirectory, .{.iterate = true}) catch |e| {
+			if(e != error.FileNotFound) {
```

Comment:
The convention is to call errors `err`. So please don't just change it for no reason.